templates/12-autosetup.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+# 12-autosetup.sh: Builds Autosetup sources
+##@copyright GPL-2.0+
+
+build_autosetup_probe(){
+	[ -e "$SRCDIR"/autosetup/autosetup ] && \
+		[ -e "$SRCDIR"/auto.def ] && \
+		[ -e "$SRCDIR"/configure ]
+}
+
+build_autosetup_configure() {
+	mkdir -vp "$BLDDIR" \
+		|| abdie "Failed to create build directory: $?."
+	cd "$BLDDIR" \
+		|| abdie "Failed to enter build directory: $?."
+
+	if [[ "x${AUTOTOOLS_TARGET[*]}" != "x" ]]
+	then
+		# $AUTOTOOLS_TARGET may be set internally by HWCAPS script.
+		# do nothing.
+		AUTOSETUP_TARGET=("${AUTOTOOLS_TARGET[@]}")
```

Comment:
Note: autosetup is basically compatible with options from `AUTOTOOLS_TARGET`, so just reuse it.